core/quantity/src/Parser.ts

```diff
@@ -474,77 +502,116 @@ export class Parser {
     return undefined;
   }
 
-  private static getQuantityValueFromParseTokens(tokens: ParseToken[], format: Format, unitsConversions: UnitConversionSpec[]): QuantityParseResult {
-    const defaultUnit = format.units && format.units.length > 0 ? format.units[0][0] : undefined;
-    // common case where single value is supplied
-    if (tokens.length === 1) {
-      if (tokens[0].isNumber) {
-        if (defaultUnit) {
-          const conversion = Parser.tryFindUnitConversion(defaultUnit.label, unitsConversions, defaultUnit);
-          if (conversion) {
-            const value = (tokens[0].value as number) * conversion.factor + conversion.offset;
-            return { ok: true, value };
-          }
-        } else {
-          // if no conversion or no defaultUnit, just return parsed number
-          return { ok: true, value: tokens[0].value as number };
-        }
-      } else {
-        // only the unit label was specified so assume magnitude of 1
-        const conversion = Parser.tryFindUnitConversion(tokens[0].value as string, unitsConversions, defaultUnit);
-        if (undefined !== conversion)
-          return { ok: true, value: conversion.factor + conversion.offset };
-        else
-          return { ok: false, error: ParseError.NoValueOrUnitFoundInString };
+  /**
+   * Get what the unit conversion is for a unitless value.
+   */
+  private static getDefaultUnitConversion(tokens: ParseToken[], unitsConversions: UnitConversionSpec[], defaultUnit?: UnitProps){
+    let unitConversion = defaultUnit ? Parser.tryFindUnitConversion(defaultUnit.label, unitsConversions, defaultUnit) : undefined;
+    if(!unitConversion){
+      // No default unit conversion, take the first valid unit.
+      const uniqueUnitLabels = [...new Set(tokens.filter((token) => token.isString).map((token) => token.value as string))];
```

Comment:
Can you add a test case that explicitly demonstrates this behavior?

Reply:
I added one, thanks